doc/specs/#5000 - Process Model 2.0/#5000 - Process Model 2.0.md

```diff
@@ -0,0 +1,1145 @@
+it shows that however we do end up implementing support for default terminal
+apps, the proposed Window/Content/Monarch/Servant architecture will still be a
+compatible solution.
+
+## UI/UX Design
+
+There's not really all that much UI for this scenario. Ideally, all this will
+happen behind the scenes, without the user really being exposed to the inner
+machinations of the Terminal processes.
+
+At the very least, resizes will occur off of the UI thread, which will help that
+scenario feel better. While resizes happen fairly responsively in Release
+builds, Debug builds used for development have fairly painful resizes occuring
+on the UI thread, blocking all input until they're finished.
+
+Ideally, we'll want the tab that's being dragged to be able to show the contents
+of the tab as we're dragging it, similar to the way that Edgium currently does.
+However, this is expected to be challenging and something that won't be a part
+of the initial release of tab tear-out.
+
+## Capabilities
+
+<table>
+<tr>
+<td><strong>Accessibility</strong></td>
+<td>
+
+When working on the implementation of the window/content process split, we'll
+need to ensure that the contents of the buffer are still readable by
+accessibility tools like Narrator and NVDA. We'll be moving the actual buffer
+out of the window process into the content process, so we'll need to make sure
+that we can hook up the `TermControlAutomationPeer` across the process boundary.
+Presumably, this is possible trivially because it's already implemented as a
+WinRT type.
+
+I'll be especially curious to make sure that
+`TermControl::OnCreateAutomationPeer` can occur off the window process's UI
+thread, because all cross-process calls will need to happen off the UI thread.
+
+</td>
+</tr>
+<tr>
+<td><strong>Security</strong></td>
+<td>
+
+As we'll be introducing a mechanism for crossing elevation boundaries, we'll
+want to be especially careful as we make these changes.
+
+Our biggest concern regarding mixed elevation was regarding the ability for a
+non-elevated process to send input to another unelevated WT window, which was
+connected to an elevated client process. With these proposed changes, we won't
+have any unelevated windows connected to unelevated processes, so that concern
+should be mitigated.
+
+Additionally, we'll probably want to make sure that the user is visually
+prompted when a connection to an elevated client is initiated. Typically, this
+is done with a UAC prompt, which seems reasonable in this scenario as well.
+
+We'll likely want to default elevated windows to spawning unelevated
+connections, as to prevent accidentally running connections as an administrator.
+This is in contrast to existing behavior, where all connections in an elevated
+WT window are elevated.
+
+Furthermore, we'll want to ensure that there's nothing that an unelevated client
+process could do to trigger any sort of input callback in the parent window. If

```

Comment:
> Furthermore, we'll want to ensure that there's nothing that an unelevated client
process could do to trigger any sort of input callback in the parent window.

How can this be realistically enforced? (Beyond just fixing holes when someone steps on them)
How do you protect against extensions introducing these vectors?
How do you remind every developer 10 years forward to keep these rules?
The point behind elevation is to have this hard separation.
The mixed elevation idea is just saying "We'll implement a backdoor, but don't worry we'll keep it secure".
This will end up with another global opt-out toggle in GPO.
Just don't build a backdoor, it's perfectly reasonable to have a fully separated stack for everything elevated.